web/src/app/workplace/workplace-routing.module.ts

```diff
@@ -1,36 +1,44 @@
 import {NgModule} from '@angular/core';
 import {RouterModule, Routes} from '@angular/router';
-import {WorkplaceComponent} from './workplace.component'
-import {TasksComponent} from 'app/workplace/tasks/tasks.component';
-import {TaskComponent} from 'app/workplace/task/task.component';
+import {MasterAndDetailComponent} from '../shared/master-and-detail/master-and-detail.component';
+import {DomainGuard} from '../guards/domain-guard';
+import {WorkplaceComponent} from './workplace.component';
+import {TaskComponent} from './task/task.component';
 import {TaskdetailsComponent} from './taskdetails/taskdetails.component';
+import {TasklistComponent} from './tasklist/tasklist.component';
 
 const routes: Routes = [
-  {
-    path: '',
-    component: WorkplaceComponent,
-    redirectTo: 'tasks',
-    pathMatch: 'full'
-  },
-  {
-    path: '',
-    component: WorkplaceComponent,
-    children: [
-      {
-        path: 'tasks',
-        component: TasksComponent
-      },
-      {
-        path: 'tasks/:id',
-        component: TaskComponent
-      },
-      {
-        path: 'tasks/taskdetail/:id',
-        component: TaskdetailsComponent
-      }
-    ]
-  }
-];
+    {
+      path: 'tasks',
+      component: MasterAndDetailComponent,
+      canActivate: [DomainGuard],
```

Comment:
Do you need DomainGuard for tasks route?

Reply:
Guards in angular are use to allow/forbidden access to routes depending on conditions, you don't need any domain check, I believe...